### PR TITLE
Fixing bug whereby a queue name with a slash in it resulted in a REST API error

### LINKED ISSUE #114: Allow to specify queue type in rabbitmq_queue module
RabbitMQ has various types of queues

- classic
- quorum
- stream (since 3.9.x)

It would be great if we could specify queue type via rabbitmq_queue module.

### PR DESCRIPTION
**Pull request by Im0**
##### SUMMARY

Fixes #114 - If a queue name was specified with a `/` in it, this resulted in a REST API error.  Also, took the chance to add a documentation example for providing `arguments`.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

rabbitmq_queue

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by Im0** [comment]
CI is failing on unrelated issue which is being worked on in #139 

@cognifloyd @odyssey4me @jgkirschbaum if anyone could take a quick look/review it would be appreciated.  Thanks :)

**Comment by Im0** [comment]
TODO: 
* tests/integration/targets/rabbitmq_publish/tasks/ubuntu.yml - unsure why this was deleted... will put back in.
* tests/integration/targets/rabbitmq_queue/tasks/ubuntu.yml - First task mentions pika... remove comment about pika.

File: changelogs/fragments/114-queue-name-escape.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - rabbitmq_queue - fixing an issue where a special character in the queue name would result in an API error
```
Comment (Andersson007) [inline changelogs/fragments/114-queue-name-escape.yml:2]:
```suggestion
  - rabbitmq_queue - fixing an issue where a special character in the queue name would result in an API error (https://github.com/ansible-collections/community.rabbitmq/issues/114).
```

**Review by Andersson007 — COMMENTED** [review]
@Im0 thanks for the bugfix!

one formatting thing from me

@csmart @odyssey4me @cognifloyd do you have any opinions on the patch?

**Comment by Im0** [comment]
Thanks @Andersson007 !  Other rabbitmq_* modules had this issue previously and appear to also have been corrected with `urllib_parse.quote(module.params['name'], '')`.  With the additional tests in place this is low risk in my opinion.  Hopefully we can get the tick from one more reviewer as I'd like to release v1.2.3 of the collection asap. :)

**Review by Andersson007 — APPROVED** [review]
@Im0 SGTM, thanks! if no objections, i think it's OK if you merge it tomorrow

**Comment by Andersson007** [comment]
@Im0 so, as there are no objections, feel free to merge

**Comment by Im0** [comment]
Thanks @Andersson007

**Comment by Andersson007** [comment]
@Im0 thanks for the contribution!
Just a fyi if once you get bored with doing only rabbitmq stuff (though there's a lot at the moment), there are many other opportunities to contribute.
Besides other collections, see the [Contributor Path](https://docs.ansible.com/ansible/devel/community/contributor_path.html) for details.
If there are any question, I'd be always happy to clarify them here or via email.
Thanks!

**Pull request merged**